### PR TITLE
Feature - Search Video

### PR DESCRIPTION
**Pull request by at-hungpham**
Description :
- Search UI
- Search Api + Load More

### CODE REVIEW

**Comment by at-hungpham** [comment]
Screen Shot 2019-10-04 at 10 50 42 AM

**Comment by at-bienle** [comment]
:+1: There are no linter violations.

File: FinalProject/View/Controller/SearchViewController/SearchViewModel.swift [hunk 1]
```diff
@@ -1,9 +1,37 @@
...
-//  Created by PCI0010 on 10/2/19.
-//  Copyright © 2019 Asiantech. All rights reserved.
-//
-
 import Foundation
+import UIKit
+
+final class SearchViewModel {
+
+    var mySearchs: [Channel] = []
```
Comment (at-bienle) [inline FinalProject/View/Controller/SearchViewController/SearchViewModel.swift]: Đổi tên

File: FinalProject/View/Custom/SearchCell/SearchCellModel.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import Foundation
+
+final class SearchCellModel {
+
+    var title: String
+    var description: String
+    var publishedAt: String
+    var imageURL: String
+
+    init(mySearch: Channel) {
```
Comment (at-bienle) [inline FinalProject/View/Custom/SearchCell/SearchCellModel.swift]: Đổi tên  `mySearch`

File: FinalProject/View/Custom/SearchCell/SearchCell.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    func updateUI() {
+        guard let viewModel = viewModel else { return }
+        titleLabel.text = viewModel.title
+        descriptionLabel.text = viewModel.description
+        publicdateLabel.text = viewModel.publishedAt
+        videoImageView.sd_setImage(with: URL(string: viewModel.imageURL))
+    }
+
+    override func awakeFromNib() {
```
Comment (at-bienle) [inline FinalProject/View/Custom/SearchCell/SearchCell.swift]: Không dùng thì xoá

File: FinalProject/View/Controller/SearchViewController/SearchViewController.swift [hunk 4]
```diff
@@ -21,5 +17,73 @@ final class SearchViewController: UIViewController {
...
+        configSearchBar()
+    }
+
+    func configSearchBar() {
+        searchController.searchResultsUpdater = self
+        searchController.dimsBackgroundDuringPresentation = false
+        searchController.definesPresentationContext = true
+        searchController.searchBar.placeholder = "Search"
+        searchController.searchBar.tintColor = UIColor.white
+        searchController.searchBar.barTintColor = UIColor.red
```
Comment (at-bienle) [inline FinalProject/View/Controller/SearchViewController/SearchViewController.swift]: Không cần `UIColor`

File: FinalProject/Model/CoreAPI/Api.Snippet.swift [hunk 5]
```diff
@@ -18,7 +18,7 @@ extension ApiManager.Snippet {
     }
 
     struct SearchResult {
-        var mySearch: [Channel] = []
+        var searchedVideo: [Channel] = []
```
Comment (at-bienle) [inline FinalProject/Model/CoreAPI/Api.Snippet.swift]: Thêm s `Videos`

File: FinalProject/View/Controller/HomeViewController/HomeViewModel.swift [hunk 6]
```diff
@@ -9,10 +9,11 @@ final class HomeViewModel {
 
     // Mark : - Load Data From API
     func loadDataTrending(completion: @escaping (Bool) -> Void) {
-        ApiManager.Snippet.getTrendingData() { (result) in
```
Comment (at-bienle) [inline FinalProject/View/Controller/HomeViewController/HomeViewModel.swift]: WARNING - When using trailing closures, empty parentheses should be avoided after the method call.

**Comment by at-bienle** [comment]
Total linter violations in pull request: 1
Serious: 0
Warnings: 1

**Comment by at-bienle** [comment]
:+1: There are no linter violations.

**Pull request merged**